### PR TITLE
Lower ONNX Erf to math library function call

### PR DESCRIPTION
**Pull request by doru1004**
Lower erf to math library function in 2 steps
1. ONNX Dialect `onnx.Erf` to Krnl Dialect `krnl.erf`
2. Krnl Dialect `krnl.erf` to LLVM IR Dialect `call erff()`

### CODE REVIEW

**Review by AlexandreEichenberger — COMMENTED** [review]
Could I ask you to add one mlir test since its the first to use this functionality, which we will hopefully reuse a lot for the other trig functions

**Comment by doru1004** [comment]
> Could I ask you to add one mlir test since its the first to use this functionality, which we will hopefully reuse a lot for the other trig functions

Absolutely, the patch is not ready hence why it's marked with WIP :)

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM except for all the "auto". Talked about it earlier, think we should be explicit when not complicated. Though fine as is, would prefer seeing fewer of them.

**Comment by doru1004** [comment]
@tjingrant is travis stuck? It's been more than 4 hours now and it hasn't finished.

File: src/Conversion/ONNXToKrnl/Math/Elementwise.cpp [hunk 1]
```diff
@@ -609,6 +615,20 @@ Value emitScalarOpFor<ONNXLessOp>(ConversionPatternRewriter &rewriter,
   }
 }
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXErfOp
+//===----------------------------------------------------------------------===//
+
+template <>
+Value emitScalarOpFor<ONNXErfOp>(ConversionPatternRewriter &rewriter,
```
Comment (tungld) [inline src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:623]: This looks redundant since we already have `struct ScalarOp<ONNXErfOp>`.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, nice use of C++

**Pull request merged**